### PR TITLE
Add help command

### LINKED ISSUE #41: Add "help" command
Consider to add a "help" command to a REPL with information about operators/functions/constants

### PR DESCRIPTION
**Pull request by pickfire**
Close #41 

cc @andy128k

Screenshot_20201102_012616

For the next thing I would probably add color to clap help message.

### CODE REVIEW

File: src/error/mod.rs [hunk 1]
```diff
@@ -25,5 +30,42 @@ pub fn handler(e: CalcError) -> String {
...
+            }
+        ),
+    }
+}
+
+/// Convert iterator into strings of chunks of 8 right padded with space.
+fn blocks(mut iter: impl Iterator<Item = &'static str> + ExactSizeIterator) -> String {
+    // calculate max width but ideally this should be calculated once
+    let mut max_width = 79; // capped at 79
+    if let Ok(s) = std::env::var("COLUMNS") {
```
Comment (pickfire) [inline src/error/mod.rs]: Environment variable seemed lost here, not sure why.

**Comment by andy128k** [comment]
Nice!

Few comments:
1. Order of operators is strange.
2. README also mentions `**`
3. README says nothing about `!`

**Comment by oppiliappan** [comment]
Looking good @pickfire.

@andy128k `**` doesn't show up because it is an alias for `^`, implemented only on the parsing end, there is no entry for the same in the `OPERATORS` hashmap. The factorial operator is missing from the readme, good catch!

File: src/error/mod.rs (hunk 1 above)
Comment (oppiliappan) [reply src/error/mod.rs]: `$COLUMNS` and `$ROWS` are shell variables and are usually available only to bash, other programs can't access them unless they are manually exported (I believe that is the reason). We could perhaps use another crate to fetch terminal size, or if that feels too contrived, simply run a `std::process::Command`: `stty size`.

Comment (pickfire) [reply src/error/mod.rs]: I got some error running `stty size`.
```
[src/error/mod.rs:41] output = Output {
    status: ExitStatus(
        ExitStatus(
            256,
        ),
    ),
    stdout: "",
    stderr: "stty: \'standard input\': Inappropriate ioctl for device\n",
}
```
```rust
            // calculate max width but ideally this should be calculated once
            let output = Command::new("stty")
                .arg("size")
                .output()
                .expect("failed to get `stty size`");
            let mut max_width = 79; // capped at 79
            if let Some(b) = dbg!(output).stdout.splitn(2, |&c| c == b' ').nth(1) {
                if let Ok(s) = std::str::from_utf8(b) {
                    if let Ok(n) = s.parse() {
                        if n < max_width {
                            max_width = n;
                        }
                    }
                }
            }
```

Comment (pickfire) [reply src/error/mod.rs]: I guess I will just use `term_size` crate for this, it also works on windows but this method does not.

**Comment by oppiliappan** [comment]
Thanks a lot for working on this @pickfire!

**Pull request merged**